test/config/utility.cc

```diff
@@ -960,8 +960,6 @@ void ConfigHelper::setPorts(const std::vector<uint32_t>& ports, bool override_po
     setConnectTimeout(std::chrono::seconds(5));
 #endif
   }
-
-  finalized_ = true;
```

Comment:
Can you explain why we need this?

Reply:
This is as in Maistra 2.2

Reply:
When I compare this full file (`test/config/utility.cc`) with what's in vanilla upstream [1] I get a bit confused, I observe more changes then I would intuitively expect.. can you double check if that is to be expected or not? 

So having said that, vanilla Envoy 1.20 *does* have this line of code, I wonder why we dropped it in our code base. So something that would be interesting to know is if we can figure out if and why we need to diverge (maybe git blame can help here?)

[1] https://raw.githubusercontent.com/envoyproxy/envoy/release/v1.20/test/config/utility.cc

Reply:
There are many changes indeed, but I have just merged all the stuff from 1.22. It's hard to say if all of them are really needed but, as far as I have understood, all changes from upstream have to be merged unless they clash with Maistra specific features (this relates mostly with OpenSSL).
As regards the modification in this specific commit, I have checked:
- In 1.20 and 2.2 that line of code (finalized_ = true;) was present, together with  timer starting;
- In 1.22 that line was removed, together with the timer.
I think in 1.22 they have rearranged the code in a way that they must be removed. In fact, if they are kept, the multi_envoy test fails.
 Then, I think we should align to 1.22 and remove also the timer.

Reply:
I have checked the code changes and differences from v1.22/2.2 through v1.22/2.3 and also 2.2->2.3 for this file.  
All differences except one are accounted for.

The outstanding change in this file in the 2.3 branch is the the finalization code was moved by Alyssa from the setPorts() to finalize().  Consequently the lines of code in setPorts() should be removed.

```
diff test/config/utility.cc ~/work/envoyproxy/envoy/test/config/utility.cc
953,964d952
<   if (!connect_timeout_set_) {
< #ifdef __APPLE__
<     // Set a high default connect timeout. Under heavy load (and in particular in CI), macOS
<     // connections can take inordinately long to complete.
<     setConnectTimeout(std::chrono::seconds(30));
< #else
<     // Set a default connect timeout.
<     setConnectTimeout(std::chrono::seconds(5));
< #endif
<   }
< 
<   finalized_ = true;
```
    
This code was moved to finalize().